_ml-commons-plugin/model-serving-framework.md

```diff
@@ -0,0 +1,250 @@
+
+### Get the `model_id`
+
+To load a model, you need the `model_id`. To find the `model_id`, take the `task_id` from the model's upload operations API response and use the `GET _ml/tasks` API.
+
+This example request uses the `task_id` from the upload example.
+
+```json
+GET /_plugins/_ml/tasks/ew8I44MBhyWuIwnfvDIH
+
+OpenSearch responds with the `model_id`. 
+
+```json
+{
+  "model_id" : "WWQI44MBbzI2oUKAvNUt", 
+  "task_type" : "UPLOAD_MODEL",
+  "function_name" : "TEXT_EMBEDDING",
+  "state" : "COMPLETED",
+  "worker_node" : "KzONM8c8T4Od-NoUANQNGg",
+  "create_time" : 3455961564003,
+  "last_update_time" : 3216361373241,
+  "is_async" : true
+}
+```
+
+### Load model from index
+
+With the `model_id`, you can now load the model from the model's index, then create an instance of the model to save in an ML node's available cache.
+
+Add the `model_id` to the load API operation. 
+
+```json
+POST /_plugins/_ml/models/<model_id>/_load
+```
+
+#### Request: Load into any available ML node
+
+In this example request, OpenSearch loads the model into any available OpenSearch node:
+
+```json
+POST /_plugins/_ml/models/WWQI44MBbzI2oUKAvNUt/_load
+```
+
+#### Request: Load into a specific node
+
+If you want to reserve the memory of other ML nodes within your cluster, you can load your model into a specific node(s) by specifying the `node_id` in the request body:
+
+```json
+POST /_plugins/_ml/models/WWQI44MBbzI2oUKAvNUt/_load
+{
+    "node_ids": ["4PLK7KJWReyX0oWKnBA8nA"]
+}
+```
+
+#### Response
+
+All models load asynchronously. Therefore, the load API responds a new `task_id` based on the load, and returns a `status` for the task.
+
+```json
+{
+  "task_id" : "hA8P44MBhyWuIwnfvTKP",
+  "status" : "CREATED"
+}
+```
+
+### Check model load status
+
+With your `task_id` from the load response, you can use the `GET _ml/tasks` API to see the loading status of your model. Before a loaded model can be used for inferences, the load tasks `state` must show as `COMPLETED`. 
+
+#### Request
+
+```json
+GET /_plugins/_ml/tasks/hA8P44MBhyWuIwnfvTKP
+```
+
+#### Response
+
+```json
+{
+  "model_id" : "WWQI44MBbzI2oUKAvNUt",
+  "task_type" : "LOAD_MODEL",
+  "function_name" : "TEXT_EMBEDDING",
+  "state" : "COMPLETED",
+  "worker_node" : "KzONM8c8T4Od-NoUANQNGg",
+  "create_time" : 1665961803150,
+  "last_update_time" : 1665961815959,
+  "is_async" : true
+}
+```
+
+## Use loaded model for inferences
+
+After the model has been loaded, you can enter the `model_id` into the [predict API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api#predict) to perform inferences.
+
+```json
+POST /_plugins/_ml/models/<model_id>/_predict
+```
+
+
+### Request
+
+```json
+POST /_plugins/_ml/models/WWQI44MBbzI2oUKAvNUt/_predict
```

Comment:
This sample request will get model from index first, then execute model. Suggest also add this sample request which will runs faster as no need to get model from index. 

```
# Sample request
POST /_plugins/_ml/_predict/text_embedding/WWQI44MBbzI2oUKAvNUt
{
  "text_docs":[ "today is sunny"],
  "return_number": true,
  "target_response": ["sentence_embedding"]
}
# Sample response
{
  "inference_results" : [
    {
      "output" : [
        {
          "name" : "sentence_embedding",
          "data_type" : "FLOAT32",
          "shape" : [
            384
          ],
          "data" : [
            -0.023315024,
            0.08975691,
            0.078479774,
            ...
          ]
        }
      ]
    }
  ]
}
```